cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -303,7 +304,8 @@ public synchronized void delete(final NamespaceId namespaceId) throws Exception
   }
 
   @Override
-  public synchronized void deleteDatasets(NamespaceId namespaceId) throws Exception {
+  @AuthEnforce(entity = "namespaceId", enforceOn = NamespaceId.class, privileges = Action.ADMIN)
+  public synchronized void deleteDatasets(@Name("namespaceId") NamespaceId namespaceId) throws Exception {
     // TODO: CDAP-870, CDAP-1427: Delete should be in a single transaction.
     if (!exists(namespaceId)) {
```

Comment:
@chtyim : The authorization enforcement line is generated as the first line in this method. I think we should be doing the existence checks before auth checks as before. In this cause should we create a deleteDatasetsHelper method which does the delete after auth check and keep the pre checks here ? Or is there a better way to do this ?